packages/flutter/lib/src/widgets/single_child_scroll_view.dart

```diff
@@ -32,6 +32,128 @@ import 'scrollable.dart';
 /// that a [SingleChildScrollView] containing a [ListBody] or [Column] with
 /// many children.
 ///
+/// ## Sample code: Using [SingleChildScrollView] with a [Column]
+///
+/// Sometimes a layout is designed around the flexible properties of a [Column],
+/// but there is the concern that on some devices, there might not be enough
+/// room to see the entire contents. In that case, it might make sense to wrap
+/// the layout in a [SingleChildScrollView].
+///
+/// Simply doing so, however, usually results in a conflict between the [Column],
+/// which typically tries to grow as big as it can, and the [SingleChildScrollView],
+/// which provides its children with an infinite amount of space.
+///
+/// To resolve this apparent conflict, there are a couple of techniques, as
+/// discussed below. These techniques should only be used when the content is
+/// normally expected to fit on the screen, so that the lazy instantiation of
+/// a sliver-based [ListView] or [CustomScrollView] is not expected to provide
+/// any performane benefit. If the viewport is expected to usually contain
+/// content beyond the dimensions of the screen, then [SingleChildScrollView]
+/// would be very expensive.
+///
+/// ### Centering, spacing, or aligning fixed-height content
+///
+/// If the content has fixed (or intrinsic) dimensions but needs to be spaced out,
+/// centered, or otherwise positioned using the [Flex] layout model of a [Column],
+/// the following technique can be used to provide the [Column] with a minimum
+/// dimension while allowing it to shrink-wrap the contents when there isn't enough
+/// room to apply these spacing or alignment needs.
+///
+/// A [LayoutBuilder] is used to obtain the size of the viewport (implicitly via
+/// the constraints that the [SingleChildScrollView] sees, since viewports
+/// typically grow to fit their maximum height constraint). Then, inside the
+/// scroll view, a [ConstrainedBox] is used to set the minimum height of the
+/// [Column], providing the [Column] with some remaining space to allocation
+/// as specified by its [Column.mainAxisAlignment]. In this example, the
+/// children are spaced out equally, unless there's no more room, in which case
+/// they stack vertically and scroll.
+///
+/// ```dart
+/// new LayoutBuilder(
+///   builder: (BuildContext context, BoxConstraints viewportConstraints) {
+///     return SingleChildScrollView(
+///       child: new ConstrainedBox(
+///         constraints: new BoxConstraints(
+///           minHeight: viewportConstraints.maxHeight,
```

Comment:
This is confusing. How are we avoiding making infinity  the column's max height constraint?

Reply:
I think I dropped a `mainAxisSize: MainAxisSize.min` somewhere while copying from my test app to the docs sample code. Oops.

Reply:
Oh, no. It's because there's no Expanded widgets, so by default the Column shrinks to fit the children.

Reply:
added some text to clarify.